### PR TITLE
Enable global invariant on OSX

### LINKED ISSUE #10329: Load ICU on demand on MAC
This is needed to have the globalization invariant feature work on MAC

### PR DESCRIPTION
**Pull request by tarekgh**
Fixes #10329

While building System.Globalization.native for OSX, we used to statically link to ICU library 

```
Users/tarek/oss/coreclr/bin/Product/OSX.x64.Release/System.Globalization.Native.dylib:
        ...
	/usr/lib/libicucore.A.dylib (compatibility version 1.0.0, current version 57.1.0)
        ...
```

The change here is instead of static linking to ICU library, we'll dynamic loading the library at run time. This will allow us to not load the ICU library at all when running in the Globalization Invariant mode.

Technically we didn't change from where we load the ICU library.

### CODE REVIEW

**Comment by tarekgh** [comment]
@jkotas @janvorli could you please have a look? 

CC @danmosemsft

**Comment by tarekgh** [comment]
@dotnet-bot help

**Comment by tarekgh** [comment]
@dotnet-bot test Ubuntu x64 corefx_baseline

File: src/corefx/System.Globalization.Native/icushim.cpp [hunk 1]
```diff
@@ -170,11 +170,31 @@ bool FindLibWithMajorMinorSubVersion(int* majorVer, int* minorVer, int* subVer)
     return false;
 }
 
-// GlobalizationNative_LoadICU
-// This method get called from the managed side during the globalization initialization. 
-// This method shouldn't get called at all if we are running in globalization invariant mode
-// return 0 if failed to load ICU and 1 otherwise
-extern "C" int32_t GlobalizationNative_LoadICU()
+#ifdef OSX_PLATFORM
```
Comment (janvorli) [inline src/corefx/System.Globalization.Native/icushim.cpp]: None of the functions / version constants above are needed on OSX, could you please ifdef them out?

File: src/corefx/System.Globalization.Native/icushim.cpp [hunk 2]
```diff
@@ -170,11 +170,31 @@ bool FindLibWithMajorMinorSubVersion(int* majorVer, int* minorVer, int* subVer)
...
+
+    // in OSX all ICU APIs exist in the same library libicucore.A.dylib 
+    libicui18n = libicuuc;
+
+    return true;
+}
+
+#else
+
+bool InitializeICULinks(char* symbolName, char* symbolVersion)
```
Comment (janvorli) [inline src/corefx/System.Globalization.Native/icushim.cpp]: Why is the name InitializeICU**Links**? I'm not sure what the "links" means. Would you mind renaming it to FindICULibs? That would represent better what the function does.

File: src/corefx/System.Globalization.Native/CMakeLists.txt [hunk 3]
```diff
@@ -32,8 +32,9 @@ else()
         message(FATAL_ERROR "Cannot find libicucore, skipping build for System.Globalization.Native. .NET globalization is not expected to function.")
         return()
     endif()
-    # On Darwin, we always use the OS provided ICU
-    SET(FEATURE_FIXED_ICU_VERSION 1)
+
+    add_definitions(-DOSX_ICU_LIBRARY_PATH=\"${ICUCORE}\")
+    add_definitions(-DOSX_PLATFORM)
```
Comment (tarekgh) [inline src/corefx/System.Globalization.Native/CMakeLists.txt]: good I'll do that

File: src/corefx/System.Globalization.Native/icushim.cpp (hunk 1 above)
Comment (tarekgh) [inline src/corefx/System.Globalization.Native/icushim.cpp]: will do that too

File: src/corefx/System.Globalization.Native/icushim.cpp (hunk 2 above)
Comment (tarekgh) [inline src/corefx/System.Globalization.Native/icushim.cpp]: I am not good in naming. I used links to include the libraries and symbols. but I can change it as you suggested as I don't have strong opinion on that :-)

**Comment by tarekgh** [comment]
@dotnet-bot test Ubuntu x64 corefx_baseline

**Comment by tarekgh** [comment]
@janvorli I have addressed all your feedback. thanks.

**Comment by tarekgh** [comment]
#10264

**Comment by tarekgh** [comment]
thanks @janvorli for your review.

**Pull request merged**